### PR TITLE
part of PR 340 moving syscalls into structure

### PR DESCRIPTION
**Pull request by tommady**
as requested by separating #340 into small PRs

this one is the second part of moving all system calls into a RootFS structure to make them mockable.
please help review 🙇🏻

### CODE REVIEW

File: src/rootfs.rs [hunk 1]
```diff
@@ -29,227 +28,461 @@ struct MountOptions<'a> {
...
-    let linux = spec.linux().as_ref().context("no linux in spec")?;
-
-    match linux.rootfs_propagation().as_deref() {
-        Some("shared") => flags |= MsFlags::MS_SHARED,
-        Some("private") => flags |= MsFlags::MS_PRIVATE,
-        Some("slave" | "unbindable") | None => flags |= MsFlags::MS_SLAVE,
-        Some(uknown) => bail!("unknown rootfs_propagation: {}", uknown),
+/// Holds information about rootfs
+pub struct RootFS {
+    command: Box<dyn Syscall>,
```
Comment (utam0k) [inline src/rootfs.rs:33]: Sorry, this s mixed up by my mistake, but let's unify the name to `syscall`.

**Comment by utam0k** [comment]
@tommady This PR just uses Syscalls so that mocks can be used, right?

File: src/rootfs.rs (hunk 1 above)
Comment (tommady) [reply src/rootfs.rs:33]: Do you mean the field name from command to syscall?
or something else, please guide.

Comment (utam0k) [reply src/rootfs.rs:33]: Let's simply change the name `command` to `syscall`.

**Comment by utam0k** [comment]
@tommady Thanks for always separating the PR from the rest. It makes it easier for me to review them :)

**Review by utam0k — APPROVED** [review]
LGTM

**Pull request merged**